planner/core/preprocess.go

```diff
@@ -819,11 +819,17 @@ func checkColumn(colDef *ast.ColumnDef) error {
 			return err
 		}
 	case mysql.TypeFloat, mysql.TypeDouble:
-		if tp.Decimal > mysql.MaxFloatingTypeScale {
-			return types.ErrTooBigScale.GenWithStackByArgs(tp.Decimal, colDef.Name.Name.O, mysql.MaxFloatingTypeScale)
-		}
-		if tp.Flen > mysql.MaxFloatingTypeWidth {
-			return types.ErrTooBigPrecision.GenWithStackByArgs(tp.Flen, colDef.Name.Name.O, mysql.MaxFloatingTypeWidth)
+		if tp.Decimal == -1 {
```

Comment:
Any information about this logic? such as MySQL's document.

Reply:
The document is [here](https://dev.mysql.com/doc/refman/8.0/en/floating-point-types.html), but its description is not detailed enough.
Use MySQL to test, we can find such constraints:

FLOAT(p) allows p from 0 to 53.
DOUBLE(p) is not allowed.
FLOAT(M,D) and DOUBLE(M,D) allows M from 0 to 255 and D from 0 to 30.

![image](https://user-images.githubusercontent.com/1070352/96529880-d1519800-12b8-11eb-9ee2-45a3643e1645.png)
PTAL @qw4990  @wjhuang2016